File: 2019/2/2.lua
#!/usr/bin/env lua
-- usage: 2/2.lua [filename]
local master = require("intcode"):load(...)
local m = master:copy()
m[1] = 12
m[2] = 2
m:run()
print(m[0])

for noun = 0, 99 do
    for verb = 0, 99 do
        m = master:copy()
        m[1] = noun
        m[2] = verb
        m:run()
        if m[0] == 19690720 then
            print(100 * noun + verb)
            return
        end
    end
end

File: 2019/2/2.pl
#!/usr/bin/env swipl
% usage: swipl 2.pl [FILENAME]

:- module(aoc2019day2, []).
:- module(aoc2019day2).
:- initialization(main, main).
:- use_module('../intcode.pl', [interpret/2, load/3]).

main([File]) :-
    phrase_from_file(load([Instr, _, _ | Rest]), File),
    interpret([Instr, 12, 2 | Rest], [A | _]),
    writeln(A),
    % 611,697,809 inferences, 40.970 CPU in 41.121 seconds (100% CPU, 14930450 Lips)
    time(interpret([Instr, X, Y | Rest], [19690720 | _])),
    format("~|~`0t~d~2+~|~`0t~d~2+", [X, Y]).

main([]) :- main(['2019/2/input.txt']).
